### ISSUE TITLE
ttkTriangulation assumes 64bit storage of topology in vtkCellArray

### ISSUE THREAD

**Issue by lhofmann** [topology-tool-kit/ttk#490]
**Describe the bug**
The VTK 9 version of vtkCellArray supports both 32bit and 64bit storage of the topology.
However, ttkTriangulation casts results of `vtkCellArray::GetConnectivityArray()` and  `vtkCellArray::GetOffsetsArray()` to LongSimplexId: https://github.com/topology-tool-kit/ttk/blob/dev/core/vtk/ttkTriangulationAlgorithm/ttkTriangulation.cpp#L322
This leads to undefined behaviour when `vtkCellArray::IsStorage64Bit()` returns false.

**To Reproduce**
```python
import vtk
import topologytoolkit as ttk

reader = vtk.vtkXMLUnstructuredGridReader()
reader.SetFileName('ttk-data/bivariateToy.vtu')
reader.Update()
# this causes the bug:
reader.GetOutput().GetCells().ConvertTo32BitStorage()

csp = ttk.ttkContinuousScatterPlot()
csp.SetInputDataObject(reader.GetOutput())
csp.Update()
```

This will result in an error similar to
```
[ExplicitTriangulation] [ERROR] Unable to create triangulation for cells of dimension 4 or higher (34359738375).
```

**Expected behavior**
It should either also work with 32bit data, or ttkTriangulation would need to check the storage type and raise an error.

**Comment by julien-tierny** [comment]
hi @lhofmann 
thanks for spotting that out.
we'll have a look into this before pulling the next release.
cheers,

**Comment by julien-tierny** [comment]
closing (addressed in #526)
